### ISSUE TITLE
Consume from multiple transports with one worker

### ISSUE THREAD

**Issue by darthf1** [petitpress/gps-messenger-bundle#25]
Hi!

Thank you for this package. I'm trying to migrate from RabbitMQ to GPS. I currently have setup multiple transports, which are consumed by one worker with the following [command](https://symfony.com/doc/current/messenger.html#prioritized-transports):
```
bin/console messenger:consume topic1 topic2 topic3
```
When I publish a message on topic2, it takes 30+ seconds to get consumed, if it even gets consumed.

When I publish a message on topic2, and I consume that topic first:
```
bin/console messenger:consume topic2 topic1 topic3
```
The message is consumed almost instantly (at least compared to the previous situation).

Is there a specific configuration needed with gps and this bundle to be able to consume messages from multiple transports via one worker?

My config (simplified):
```php
<?php

declare(strict_types=1);

use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;

return static function (ContainerConfigurator $containerConfigurator): void {
    $containerConfigurator->extension('framework', [
        'messenger' => [
           ...
            'transports' => [
                'topic1' => [
                    'dsn' => '%env(MESSENGER_TRANSPORT_DSN)%',
                    'options' => [
                        'topic' => [
                            'name' => 'topic1',
                        ],
                        'client_config' => [
                            'hasEmulator' => '%env(bool:HAS_PUBSUB_EMULATOR)%',
                            'emulatorHost' => '%env(PUBSUB_EMULATOR_HOST)%',
                        ],
                    ],
                ],
                'topic2' => [
                    'dsn' => '%env(MESSENGER_TRANSPORT_DSN)%',
                    'options' => [
                        'topic' => [
                            'name' => 'topic2',
                        ],
                        'client_config' => [
                            'hasEmulator' => '%env(bool:HAS_PUBSUB_EMULATOR)%',
                            'emulatorHost' => '%env(PUBSUB_EMULATOR_HOST)%',
                        ],
                    ],
                ],
                'topic3' => [
                    'dsn' => '%env(MESSENGER_TRANSPORT_DSN)%',
                    'options' => [
                        'topic' => [
                            'name' => 'topic3',
                        ],
                        'client_config' => [
                            'hasEmulator' => '%env(bool:HAS_PUBSUB_EMULATOR)%',
                            'emulatorHost' => '%env(PUBSUB_EMULATOR_HOST)%',
                        ],
                    ],
                ],
            ],
        ],
    ]);
};
```

**Comment by jon-ht** [comment]
Hi, I guess this is related to how GPS consumes messages.

By default it will wait until a message is received on your `topic1`, blocking any messages sent to `topic2` and `topic3`.

There is a way to fix this by setting `returnImmediately: true` when pulling from a subscription topic.

I made a PR for this in https://github.com/petitpress/gps-messenger-bundle/pull/23, my use case was for batch messages but you might be in the same situation.

Could give it a try ?

**Comment by darthf1** [comment]
I just tried, but unfortunately it doesn't do anything for me. Do i need to explicit configure something?

```json
{
    "repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/jon-ht/gps-messenger-bundle.git"
        }
    ],
    "require": {
        "petitpress/gps-messenger-bundle": "dev-feat-subscription-pull-options",
    }
}

**Comment by jon-ht** [comment]
I'm glad it worked ! 

For your exception about `subscription[pull]` it's intended, because this option only exists on my PR.

@marforon Could you have a look at PR https://github.com/petitpress/gps-messenger-bundle/pull/23 to fix the issue ?

**Comment by marforon** [comment]
thanks @jon-ht for fix, release in [1.6.0](https://github.com/petitpress/gps-messenger-bundle/releases/tag/1.6.0 )
